src/lib/core/components/Form/hooks/useIntegrationFF.tsx

```diff
@@ -64,11 +68,11 @@ export const useIntegrationFF = (store: DynamicFieldStore, withoutDebounce?: boo
 
     React.useEffect(() => {
         return () => {
-            if (store.name) {
+            if (store.name && destroyOnUnregister) {
                 form.change(store.name, undefined);
             }
         };
-    }, []);
+    }, [destroyOnUnregister]);
```

Comment:
here this flag should not cause an effect, its value will be relevant in the callback, this is the main thing that we need

Reply:
> Its value will be relevant in the callback

I don't think so, on "unmount" it will have the first value provided in this case, and it will be not updated on prop change.

But generally, yes, the "logic part" shouldn't be triggered by its value change. And most likely it won't be changed.

Reply:
```
I don't think so, on "unmount" it will have the first value provided in this case, and it will be not updated on prop change.
```

Oh, you're right, we'll have to go back to this place and make the flag and store.name the callback was relevant